### PR TITLE
wxString::ToAscii pass parameter for replacement char

### PR DESCRIPTION
**Pull request by mtangoo**
wxString::ToAscii  replaces Unicode chars with underscore. I have a use case where the character should be something than underscore. This PR adds that feature. I tested its compiling fine

### CODE REVIEW

File: include/wx/string.h [hunk 1]
```diff
@@ -1556,13 +1556,13 @@ class WXDLLIMPEXP_BASE wxString
     static wxString FromAscii(const char *ascii, size_t len);
     static wxString FromAscii(const char *ascii);
     static wxString FromAscii(char ascii);
-    const wxScopedCharBuffer ToAscii() const;
+    const wxScopedCharBuffer ToAscii(const char replaceWith='_') const;
```
Comment (vadz) [inline include/wx/string.h]: It doesn't make sense to use top-level `const` for the parameter and we don't do it anywhere else (notably in `FromAscii()` just above!), could you please remove it, here and in the documentation?

File: interface/wx/string.h [hunk 2]
```diff
@@ -743,13 +743,16 @@ class wxString
         (underscore) character.
 
         Use mb_str() or utf8_str() to convert to other encodings.
+        
+        @param replaceWith
+            The Character that will replace found unicode character. defaults to underscore "_"
```
Comment (vadz) [inline interface/wx/string.h]: Could you please lower case "Character", upper case "defaults" and add "This parameter is new since wxWidgets 3.1.0."?

**Comment by vadz** [comment]
No objections to applying this, but please add at least a simple test for the new feature to ensure that it doesn't get broken in the future.

Also, if possible, please squash all the commits together (but I can also do it myself).

**Comment by mtangoo** [comment]
Test you mean I use it somewhere? I can do that in few minutes

File: interface/wx/string.h (hunk 2 above)
Comment (mtangoo) [inline interface/wx/string.h]: I will add that!

File: include/wx/string.h (hunk 1 above)
Comment (mtangoo) [inline include/wx/string.h]: Yes pls. I will do it

**Comment by vadz** [comment]
By "test" I mean add a unit test for it somewhere under `tests/strings`, e.g. in [this file](https://github.com/wxWidgets/wxWidgets/blob/master/tests/strings/unicode.cpp).

**Comment by mtangoo** [comment]
Checking that file, there is already a test that should work fine with default
```cpp 
CPPUNIT_ASSERT( strcmp( s.ToAscii() , msg )
```

The only addition I can add to that is putting a non default  argument, something like this one
```cpp
#define TEST_TO_FROM_ASCII(txt, replaceWith)           \
{                                                       \
    static const char *msg = txt;                        \
    static const char rw = replaceWith;                  \
    wxString s = wxString::FromAscii(msg);               \
    CPPUNIT_ASSERT( strcmp( s.ToAscii(rw) , msg ) == 0 );  \
}

TEST_TO_FROM_ASCII( "Hello, world!" , '+');
TEST_TO_FROM_ASCII( "additional \" special \t test \\ component \n :-)", '+' );
```

is this what you are saying

**Comment by vadz** [comment]
Just add another `CPPUNIT_ASSERT_EQUAL()` to `UnicodeTestCase::ToFromAscii()`...

**Comment by mtangoo** [comment]
Am not Unit test maverick. Can you comment on the code above if it correct?
I will then add that code to ```cpp CPPUNIT_ASSERT_EQUAL()```

**Pull request closed without merging**